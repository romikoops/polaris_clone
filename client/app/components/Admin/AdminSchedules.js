import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Admin.scss';
import { AdminScheduleLine } from './';
import AdminScheduleGenerator from './AdminScheduleGenerator';
import Select from 'react-select';
import '../../styles/select-css-custom.css';
// import {v4} from 'node-uuid';
import styled from 'styled-components';
import FileUploader from '../FileUploader/FileUploader';
import {RoundButton} from '../RoundButton/RoundButton';
import {v4} from 'node-uuid';
export class AdminSchedules extends Component {
    constructor(props) {
        super(props);
        console.log(props);
        this.state = {
            showList: false,
            filters: {
                hub: false,
                mot: false,
                sort: false
            },
            motFilter: {value: false, label: false},
            sortFilter: {value: false, label: false},
            hubFilter: {value: false, label: false},
        };
        this.toggleView = this.toggleView.bind(this);
        this.setMoTFilter = this.setMoTFilter.bind(this);
        this.setSortFilter = this.setSortFilter.bind(this);
        this.setHubFilter = this.setHubFilter.bind(this);
    }
    toggleView() {
        this.setState({showList: !this.state.showList});
    }
    setMoTFilter(mot) {
        if (!mot) {
            this.setState({
                motFilter: {value: false, label: false},
                filters: {
                    ...this.state.filters,
                    mot: false
                }
            });
        } else {
            this.setState({
                motFilter: mot,
                filters: {
                    ...this.state.filters,
                    mot: true
                }
            });
        }
    }
    setSortFilter(sorter) {
        if (!sorter) {
            this.setState({
                sortFilter: {value: false, label: false},
                filters: {
                    ...this.state.filters,
                    sort: false
                }
            });
        } else {
            this.setState({
                sortFilter: sorter,
                filters: {
                    ...this.state.filters,
                    sort: true
                }
            });
        }
    }
    setHubFilter(hub) {
        if (!hub) {
            this.setState({
                hubFilter: {value: false, label: false},
                filters: {
                    ...this.state.filters,
                    hub: false
                }
            });
        } else {
            this.setState({
                hubFilter: hub,
                filters: {
                    ...this.state.filters,
                    hub: true
                }
            });
        }
    }

    dynamicSort(property) {
        let sortOrder = 1;
        let prop;
        if(property[0] === '-') {
            sortOrder = -1;
            prop = property.substr(1);
        } else {
            prop = property;
        }
        return function(a, b) {
            const result1 = a[prop] < b[prop] ? -1 : a[prop] > b[prop];
            const result2 = result1 ? 1 : 0;
            return result2 * sortOrder;
        };
    }
    render() {
        const {theme, hubs, scheduleData} = this.props;
        const {filters, hubFilter, motFilter, sortFilter} = this.state;
        if (!scheduleData || !hubs) {
            return '';
        }
        const filterMoTOptions = [
            {value: 'rail', label: 'Rail'},
            {value: 'air', label: 'Air'},
            {value: 'ocean', label: 'Ocean'}
        ];
        const filterSortOptions = [
            {value: 'eta', label: 'ETA'},
            {value: 'etd', label: 'ETD'}
        ];
        const {routes, air, train, ocean} = scheduleData;
        const { showList } = this.state;
        const trainUrl = '/admin/train_schedules/process_csv';
        const vesUrl = '/admin/vessel_schedules/process_csv';
        const airUrl = '/admin/air_schedules/process_csv';
        const truckUrl = '/admin/truck_schedules/process_csv';
        const schedArr = [];
        const allSchedules = [...air, ...ocean, ...train];
        const hFilterVal = parseInt(hubFilter.value, 10);
        allSchedules.forEach(sched => {
            const schedKeys = sched.hub_route_key.split('-');
            if (!filters.hub && filters.mot && sched.mode_of_transport === motFilter.value) {
                schedArr.push(<AdminScheduleLine key={v4()} schedule={sched} hubs={hubs} theme={theme}/>);
            }
            if (filters.hub && !filters.mot && schedKeys[1] === String(hFilterVal)) {
                schedArr.push(<AdminScheduleLine key={v4()} schedule={sched} hubs={hubs} theme={theme}/>);
            }
            if (filters.hub && !filters.mot && schedKeys[0] === String(hFilterVal)) {
                schedArr.push(<AdminScheduleLine key={v4()} schedule={sched} hubs={hubs} theme={theme}/>);
            }
            if (filters.hub && filters.mot && sched.mode_of_transport === motFilter.value && schedKeys[0] === String(hFilterVal)) {
                schedArr.push(<AdminScheduleLine key={v4()} schedule={sched} hubs={hubs} theme={theme}/>);
            }
            if (filters.hub && filters.mot && sched.mode_of_transport === motFilter.value && schedKeys[1] === String(hFilterVal)) {
                schedArr.push(<AdminScheduleLine key={v4()} schedule={sched} hubs={hubs} theme={theme}/>);
            }
            if (!filters.hub && !filters.mot) {
                schedArr.push(<AdminScheduleLine key={v4()} schedule={sched} hubs={hubs} theme={theme}/>);
            }
        });
        if (filters.sort) {
            schedArr.sort(this.dynamicSort(sortFilter.value));
        }
        const StyledSelect = styled(Select)`
            .Select-control {
                background-color: #F9F9F9;
                box-shadow: 0 2px 3px 0 rgba(237,234,234,0.5);
                border: 1px solid #F2F2F2 !important;
            }
            .Select-menu-outer {
                box-shadow: 0 2px 3px 0 rgba(237,234,234,0.5);
                border: 1px solid #F2F2F2;
            }
            .Select-value {
                background-color: #F9F9F9;
                border: 1px solid #F2F2F2;
            }
            .Select-option {
                background-color: #F9F9F9;
            }
        `;
        const hubList = [];
        Object.keys(hubs).forEach(key => {
            hubList.push({value: key, label: hubs[key].data.name});
        });
        const listView = (
            <div className="layout-row flex-100 layout-wrap layout-align-start-center">
                <div className="flex-100 layout-row layout-align-start-center">
                    <div className="flex-33 layout-row layout-align-start-center">
                        <StyledSelect
                            name="mot-filter"
                            className={`${styles.select}`}
                            value={this.state.motFilter}
                            options={filterMoTOptions}
                            onChange={this.setMoTFilter}
                        />
                    </div>
                    <div className="flex-33 layout-row layout-align-start-center">
                        <StyledSelect
                            name="sort-filter"
                            className={`${styles.select}`}
                            value={this.state.sortFilter}
                            options={filterSortOptions}
                            onChange={this.setSortFilter}
                        />
                    </div>
                    <div className="flex-33 layout-row layout-align-start-center">
                        <StyledSelect
                            name="hub-filter"
                            className={`${styles.select}`}
                            value={this.state.hubFilter}
                            options={hubList}
                            onChange={this.setHubFilter}
                        />
                    </div>
                </div>
                {schedArr}
            </div>
        );

        const textStyle = {
            background: theme && theme.colors ? '-webkit-linear-gradient(left, ' + theme.colors.primary + ',' + theme.colors.secondary + ')' : 'black'
        };


        const genView = (
            <div className="layout-row flex-100 layout-wrap layout-align-start-center">
                <div className="layout-row flex-100 layout-wrap layout-align-start-center">
                    <div className={`flex-100 layout-row layout-align-space-between-center ${styles.sec_header}`}>
                        <p className={` ${styles.sec_header_text} flex-none`}  >Excel Uploads</p>
                    </div>
                    <div className={`flex-50 layout-row layout-align-space-between-center ${styles.sec_upload}`}>
                        <p className="flex-80">Upload Train Schedules Sheet</p>
                        <FileUploader theme={theme} url={trainUrl} type="xlsx" text="Train Schedules .xlsx"/>
                    </div>
                    <div className={`flex-50 layout-row layout-align-space-between-center ${styles.sec_upload}`}>
                        <p className="flex-80">Upload Air Schedules Sheet</p>
                        <FileUploader theme={theme} url={airUrl} type="xlsx" text="Air Schedules .xlsx"/>
                    </div>
                    <div className={`flex-50 layout-row layout-align-space-between-center ${styles.sec_upload}`}>
                        <p className="flex-80">Upload Vessel Schedules Sheet</p>
                        <FileUploader theme={theme} url={vesUrl} type="xlsx" text="Vessel Schedules .xlsx"/>
                    </div>
                    <div className={`flex-50 layout-row layout-align-space-between-center ${styles.sec_upload}`}>
                        <p className="flex-80">Upload Trucking Schedules Sheet</p>
                        <FileUploader theme={theme} url={truckUrl} type="xlsx" text="Truck Schedules .xlsx"/>
                    </div>
                </div>
                <AdminScheduleGenerator theme={theme} routes={routes} hubs={hubList} />
            </div>
        );

        const currView = showList ? listView : genView;

        const backButton = (<RoundButton theme={theme} text="Back to list" size="small" back iconClass="fa-th-list" handleNext={this.toggleView} />);
        const newButton = (<RoundButton theme={theme} text="New Upload" active size="small" iconClass="fa-plus" handleNext={this.toggleView} />);
        return(
            <div className="flex-100 layout-row layout-wrap layout-align-start-start">
                <div className={`flex-100 layout-row layout-align-space-between-center ${styles.sec_title}`}>
                    <p className={` ${styles.sec_title_text} flex-none`} style={textStyle} >Schedules</p>
                    { showList ? newButton : backButton }
                </div>
                {currView}
            </div>
        );
    }
}
AdminSchedules.propTypes = {
    theme: PropTypes.object,
    hubs: PropTypes.object,
    scheduleData: PropTypes.object
};
