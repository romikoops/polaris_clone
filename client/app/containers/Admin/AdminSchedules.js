import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Admin.scss';
// import { AdminPricePanel } from './AdminPricePanel';
// import {v4} from 'node-uuid';
import FileUploader from '../../components/FileUploader/FileUploader';
export class AdminSchedules extends Component {
    constructor(props) {
        super(props);
        this.state = {
            currentView: 'ocean'
        };
        this.toggleView = this.toggleView.bind(this);
    }
    toggleView(view) {
        this.setState({currentView: view});
    }
    render() {
        const {theme, hubs, schedules } = this.props;
        const { currentView } = this.state;
        // const {pricings, routes} = pricingData;
        const hubHash = {};
        hubs.forEach((hub) => {
            hubHash[hub.data.id] = hub;
        });
        console.log(schedules);
        // if (charges && hubs) {
        //     charges.forEach((charge) =>{
        //         console.log(charge.hub_id);
        //         console.log(hubHash[charge.hub_id]);
        //         if (hubHash[charge.hub_id]) {
        //             chargeList.push(<AdminChargePanel key={v4()} hub={hubHash[charge.hub_id]} theme={theme} charge={charge}/>);
        //         }
        //     });
        // }
        const trainUrl = '/admin/train_schedules/process_csv';
        const vesUrl = '/admin/vessel_schedules/process_csv';
        const airUrl = '/admin/air_schedules/process_csv';
        const railView = (
            <div className="layout-row flex-100 layout-wrap layout-align-start-center">
                <div className={`flex-100 layout-row layout-align-space-between-center ${styles.sec_upload}`}>
                    <p className="flex-none">Upload Train Schedules Sheet</p>
                    <FileUploader theme={theme} url={trainUrl} type="xlsx" text="Train Schedules .xlsx"/>
                </div>
                <div className="layout-row flex-100 layout-wrap layout-align-start-center">
                </div>
            </div>
        );
        const airView = (
            <div className="layout-row flex-100 layout-wrap layout-align-start-center">
                <div className={`flex-100 layout-row layout-align-space-between-center ${styles.sec_upload}`}>
                    <p className="flex-none">Upload Air Schedules Sheet</p>
                    <FileUploader theme={theme} url={airUrl} type="xlsx" text="Air Schedules .xlsx"/>
                </div>
                <div className="layout-row flex-100 layout-wrap layout-align-start-center">
                </div>
            </div>
        );
        const oceanView = (
            <div className="layout-row flex-100 layout-wrap layout-align-start-center">
                <div className={`flex-100 layout-row layout-align-space-between-center ${styles.sec_upload}`}>
                    <p className="flex-none">Upload Vessel Schedules Sheet</p>
                    <FileUploader theme={theme} url={vesUrl} type="xlsx" text="Vessel Schedules .xlsx"/>
                </div>
                <div className="layout-row flex-100 layout-wrap layout-align-start-center">
                </div>
            </div>
        );
        let currView = '';
        switch(currentView) {
            case 'ocean':
                currView = oceanView;
                break;
            case 'air':
                currView = airView;
                break;
            case 'rail':
                currView = railView;
                break;
            default:
                currView = oceanView;
                break;
        }
        const textStyle = {
            background: theme && theme.colors ? '-webkit-linear-gradient(left, ' + theme.colors.primary + ',' + theme.colors.secondary + ')' : 'black'
        };
        return(
            <div className="flex-100 layout-row layout-wrap layout-align-start-start">
                <div className="layout-row flex-100 layout-wrap layout-align-start-center">
                    <div className={`flex-33 layout-row layout-align-center-center ${styles.sched_selector}`} onClick={() => this.toggleView('ocean')}>
                        <i className="fa fa-ship flex-none" style={textStyle}></i>
                        <p className="flex-none flex-offset-5"  style={textStyle}> Ocean </p>
                    </div>
                    <div className={`flex-33 layout-row layout-align-center-center ${styles.sched_selector}`} onClick={() => this.toggleView('rails')}>
                        <i className="fa fa-train flex-none" style={textStyle}></i>
                        <p className="flex-none flex-offset-5"  style={textStyle}> Rail </p>
                    </div>
                    <div className={`flex-33 layout-row layout-align-center-center ${styles.sched_selector}`} onClick={() => this.toggleView('air')}>
                        <i className="fa fa-plane flex-none" style={textStyle}></i>
                        <p className="flex-none flex-offset-5"  style={textStyle}> Air </p>
                    </div>
                </div>
                <div className={`flex-100 layout-row layout-align-start-center ${styles.sec_title}`}>
                    <p className="flex-none" style={textStyle}>pricings</p>
                </div>

                {currView}
            </div>
        );
    }
}
AdminSchedules.propTypes = {
    theme: PropTypes.object,
    hubs: PropTypes.array,
    pricings: PropTypes.array
};
