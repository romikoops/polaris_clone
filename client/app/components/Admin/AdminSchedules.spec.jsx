import * as React from 'react'
import { shallow } from 'enzyme'
import {
  theme, identity, user, internalUser
} from '../../mocks'

import AdminSchedules from './AdminSchedules'

jest.mock('../../helpers', () => ({
  filters: x => x,
  capitalize: x => x
}))
jest.mock('react-redux', () => ({
  connect: (mapStateToProps, mapDispatchToProps) => Component => Component
}))

const hub = {
  data: {
    name: 'FOO_NAME',
    hub_code: 'FOO_HUB_CODE'
  },
  name: 'FOO_HUB_NAME'
}
const secondHub = {
  data: {
    name: 'BAR_NAME',
    hub_code: 'BAR_HUB_CODE'
  },
  name: 'BAR_HUB_NAME'
}
const propsBase = {
  theme,
  schedule: {
    id: 555777,
    eta: '2018-12-01T12:14:08+01:00',
    hub_route_key: '0-1'
  },
  hubs: [hub, secondHub],
  scheduleData: {
    routes: [],
    mapData: [],
    detailedItineraries: [],
    itineraryIds: [],
    itineraries: [{ modes_of_transport: 'air' }]
  },
  document: { viewer: null },
  itineraries: { filter: identity },
  adminDispatch: identity,
  setCurrentUrl: identity,
  documentDispatch: {},
  scope: {
    modes_of_transport: { air: 'Air' }
  },
  match: {
    url: 'URL'
  },
  user
}

test('shallow render', () => {
  expect(shallow(<AdminSchedules {...propsBase} />)).toMatchSnapshot()
})

test('state.showList is false', () => {
  const wrapper = shallow(<AdminSchedules {...propsBase} />)
  wrapper.setState({ showList: false })
  expect(wrapper).toMatchSnapshot()
})

test('scheduleData is falsy', () => {
  const props = {
    ...propsBase,
    scheduleData: null
  }
  expect(shallow(<AdminSchedules {...props} />)).toMatchSnapshot()
})

test('document.viewer is truthy', () => {
  const props = {
    ...propsBase,
    document: { viewer: {} }
  }
  expect(shallow(<AdminSchedules {...props} />)).toMatchSnapshot()
})

test('user.internal is truthy', () => {
  const props = {
    ...propsBase,
    user: internalUser
  }
  expect(shallow(<AdminSchedules {...props} />)).toMatchSnapshot()
})
