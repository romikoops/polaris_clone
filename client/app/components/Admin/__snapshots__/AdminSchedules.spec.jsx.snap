// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`document.viewer is truthy 1`] = `
<GenericError
  t={[Function]}
  theme={
    Object {
      "colors": Object {
        "primary": "#333",
        "secondary": "#fafafa",
      },
    }
  }
>
  <div
    className="flex-100 layout-row layout-wrap layout-align-space-around-start extra_padding_left"
  >
    <AdminUploadsSuccess
      closeDialog={[Function]}
      data={Object {}}
      t={[Function]}
      theme={
        Object {
          "colors": Object {
            "primary": "#333",
            "secondary": "#fafafa",
          },
        }
      }
    />
    <div
      className="component_view flex layout-row layout-align-start-start"
    >
      <div
        className="flex-100 layout-row layout-align-center-start"
      >
        <Tabs
          defaultActiveTabIndex={0}
          paddingFixes={true}
          wrapperTabs="layout-row flex-45 flex-sm-40 flex-xs-80"
        >
          <Tab
            extraClick={[Function]}
            icon=""
            isActive={false}
            isUniq={false}
            linkClassName=""
            mot="air"
            onClick={null}
            tabIndex={0}
            tabTitle="air"
            theme={
              Object {
                "colors": Object {
                  "primary": "#333",
                  "secondary": "#fafafa",
                },
              }
            }
          >
            <CardRoutesIndex
              adminDispatch={[Function]}
              handleClick={[Function]}
              hubs={Array []}
              itineraries={Array []}
              limit={9}
              mot="air"
              newText=""
              scope={
                Object {
                  "modes_of_transport": Object {
                    "air": "Air",
                  },
                }
              }
              sideMenuNodes={
                Array [
                  <SideOptionsBox
                    content={
                      <div
                        className="flex-100 layout-row layout-wrap layout-align-center-start"
                      >
                        
                        <CollapsingBar
                          collapsed={true}
                          content={
                            <div>
                              <div
                                className="action_section flex-100 layout-row layout-wrap layout-align-center-center"
                              >
                                <p
                                  className="flex-100"
                                >
                                  Download Schedules Sheet
                                </p>
                                <DocumentsDownloader
                                  disabled={false}
                                  documentDispatch={null}
                                  downloadUrls={Object {}}
                                  loading={false}
                                  options={Object {}}
                                  size="small"
                                  square={false}
                                  t={[Function]}
                                  target="schedules"
                                  theme={
                                    Object {
                                      "colors": Object {
                                        "primary": "#333",
                                        "secondary": "#fafafa",
                                      },
                                    }
                                  }
                                  tooltip=""
                                />
                              </div>
                            </div>
                          }
                          contentHeader=""
                          faClass="fa fa-cloud-download"
                          handleCollapser={[Function]}
                          headerWrapClasses=""
                          hideIcon={false}
                          mainWrapperStyle={Object {}}
                          minHeight=""
                          parentClass=""
                          showArrow={true}
                          showArrowSpacer={false}
                          startCollapsed={false}
                          text="Download Data"
                          theme={
                            Object {
                              "colors": Object {
                                "primary": "#333",
                                "secondary": "#fafafa",
                              },
                            }
                          }
                          wrapperContentClasses=""
                        />
                        
                      </div>
                    }
                    flexOptions=""
                    header="Data manager"
                  />,
                ]
              }
              t={[Function]}
              theme={
                Object {
                  "colors": Object {
                    "primary": "#333",
                    "secondary": "#fafafa",
                  },
                }
              }
              toggleNew={null}
              user={
                Object {
                  "company_name": "FOO_COMPANY",
                  "currency": "EUR",
                  "email": "foo@bar.baz",
                  "first_name": "John",
                  "guest": false,
                  "last_name": "Doe",
                  "phone": "6345789",
                  "role": Object {
                    "name": "shipper",
                  },
                  "role_id": 1,
                }
              }
            />
          </Tab>
        </Tabs>
      </div>
    </div>
  </div>
</GenericError>
`;

exports[`scheduleData is falsy 1`] = `""`;

exports[`shallow render 1`] = `
<GenericError
  t={[Function]}
  theme={
    Object {
      "colors": Object {
        "primary": "#333",
        "secondary": "#fafafa",
      },
    }
  }
>
  <div
    className="flex-100 layout-row layout-wrap layout-align-space-around-start extra_padding_left"
  >
    <div
      className="component_view flex layout-row layout-align-start-start"
    >
      <div
        className="flex-100 layout-row layout-align-center-start"
      >
        <Tabs
          defaultActiveTabIndex={0}
          paddingFixes={true}
          wrapperTabs="layout-row flex-45 flex-sm-40 flex-xs-80"
        >
          <Tab
            extraClick={[Function]}
            icon=""
            isActive={false}
            isUniq={false}
            linkClassName=""
            mot="air"
            onClick={null}
            tabIndex={0}
            tabTitle="air"
            theme={
              Object {
                "colors": Object {
                  "primary": "#333",
                  "secondary": "#fafafa",
                },
              }
            }
          >
            <CardRoutesIndex
              adminDispatch={[Function]}
              handleClick={[Function]}
              hubs={Array []}
              itineraries={Array []}
              limit={9}
              mot="air"
              newText=""
              scope={
                Object {
                  "modes_of_transport": Object {
                    "air": "Air",
                  },
                }
              }
              sideMenuNodes={
                Array [
                  <SideOptionsBox
                    content={
                      <div
                        className="flex-100 layout-row layout-wrap layout-align-center-start"
                      >
                        
                        <CollapsingBar
                          collapsed={true}
                          content={
                            <div>
                              <div
                                className="action_section flex-100 layout-row layout-wrap layout-align-center-center"
                              >
                                <p
                                  className="flex-100"
                                >
                                  Download Schedules Sheet
                                </p>
                                <DocumentsDownloader
                                  disabled={false}
                                  documentDispatch={null}
                                  downloadUrls={Object {}}
                                  loading={false}
                                  options={Object {}}
                                  size="small"
                                  square={false}
                                  t={[Function]}
                                  target="schedules"
                                  theme={
                                    Object {
                                      "colors": Object {
                                        "primary": "#333",
                                        "secondary": "#fafafa",
                                      },
                                    }
                                  }
                                  tooltip=""
                                />
                              </div>
                            </div>
                          }
                          contentHeader=""
                          faClass="fa fa-cloud-download"
                          handleCollapser={[Function]}
                          headerWrapClasses=""
                          hideIcon={false}
                          mainWrapperStyle={Object {}}
                          minHeight=""
                          parentClass=""
                          showArrow={true}
                          showArrowSpacer={false}
                          startCollapsed={false}
                          text="Download Data"
                          theme={
                            Object {
                              "colors": Object {
                                "primary": "#333",
                                "secondary": "#fafafa",
                              },
                            }
                          }
                          wrapperContentClasses=""
                        />
                        
                      </div>
                    }
                    flexOptions=""
                    header="Data manager"
                  />,
                ]
              }
              t={[Function]}
              theme={
                Object {
                  "colors": Object {
                    "primary": "#333",
                    "secondary": "#fafafa",
                  },
                }
              }
              toggleNew={null}
              user={
                Object {
                  "company_name": "FOO_COMPANY",
                  "currency": "EUR",
                  "email": "foo@bar.baz",
                  "first_name": "John",
                  "guest": false,
                  "last_name": "Doe",
                  "phone": "6345789",
                  "role": Object {
                    "name": "shipper",
                  },
                  "role_id": 1,
                }
              }
            />
          </Tab>
        </Tabs>
      </div>
    </div>
  </div>
</GenericError>
`;

exports[`state.showList is false 1`] = `
<GenericError
  t={[Function]}
  theme={
    Object {
      "colors": Object {
        "primary": "#333",
        "secondary": "#fafafa",
      },
    }
  }
>
  <div
    className="flex-100 layout-row layout-wrap layout-align-space-around-start extra_padding_left"
  >
    <div
      className="component_view flex layout-row layout-align-start-start"
    >
      <div
        className="layout-row flex-100 layout-wrap layout-align-start-center"
      >
        <AdminScheduleGenerator
          hubs={Array []}
          itineraries={
            Array [
              Object {
                "modes_of_transport": "air",
              },
            ]
          }
          itinerary={Object {}}
          t={[Function]}
          theme={
            Object {
              "colors": Object {
                "primary": "#333",
                "secondary": "#fafafa",
              },
            }
          }
          toggleNew={[Function]}
          vehicleTypes={Array []}
        />
      </div>
    </div>
  </div>
</GenericError>
`;

exports[`user.internal is truthy 1`] = `
<GenericError
  t={[Function]}
  theme={
    Object {
      "colors": Object {
        "primary": "#333",
        "secondary": "#fafafa",
      },
    }
  }
>
  <div
    className="flex-100 layout-row layout-wrap layout-align-space-around-start extra_padding_left"
  >
    <div
      className="component_view flex layout-row layout-align-start-start"
    >
      <div
        className="flex-100 layout-row layout-align-center-start"
      >
        <Tabs
          defaultActiveTabIndex={0}
          paddingFixes={true}
          wrapperTabs="layout-row flex-45 flex-sm-40 flex-xs-80"
        >
          <Tab
            extraClick={[Function]}
            icon=""
            isActive={false}
            isUniq={false}
            linkClassName=""
            mot="air"
            onClick={null}
            tabIndex={0}
            tabTitle="air"
            theme={
              Object {
                "colors": Object {
                  "primary": "#333",
                  "secondary": "#fafafa",
                },
              }
            }
          >
            <CardRoutesIndex
              adminDispatch={[Function]}
              handleClick={[Function]}
              hubs={Array []}
              itineraries={Array []}
              limit={9}
              mot="air"
              newText=""
              scope={
                Object {
                  "modes_of_transport": Object {
                    "air": "Air",
                  },
                }
              }
              sideMenuNodes={
                Array [
                  <SideOptionsBox
                    content={
                      <div
                        className="flex-100 layout-row layout-wrap layout-align-center-start"
                      >
                        <CollapsingBar
                          collapsed={true}
                          content={
                            <div>
                              <div
                                className="action_section flex-100 layout-row layout-align-center-center layout-wrap"
                              >
                                <p
                                  className="flex-80"
                                >
                                  Generator Sheet
                                </p>
                                <FileUploader
                                  dispatchFn={[Function]}
                                  formClasses=""
                                  size="small"
                                  square={false}
                                  t={[Function]}
                                  text="Generator Sheet"
                                  theme={
                                    Object {
                                      "colors": Object {
                                        "primary": "#333",
                                        "secondary": "#fafafa",
                                      },
                                    }
                                  }
                                  tooltip=""
                                  type="xlsx"
                                  uploadFn={null}
                                />
                              </div>
                              <div
                                className="action_section flex-100 layout-row layout-align-center-center layout-wrap"
                              >
                                <p
                                  className="flex-80"
                                >
                                  Upload Air Schedules Sheet
                                </p>
                                <FileUploader
                                  dispatchFn={[Function]}
                                  formClasses=""
                                  size="small"
                                  square={false}
                                  t={[Function]}
                                  text="Air Schedules .xlsx"
                                  theme={
                                    Object {
                                      "colors": Object {
                                        "primary": "#333",
                                        "secondary": "#fafafa",
                                      },
                                    }
                                  }
                                  tooltip=""
                                  type="xlsx"
                                  uploadFn={null}
                                />
                              </div>
                              <div
                                className="action_section flex-100 layout-row layout-align-center-center layout-wrap"
                              >
                                <p
                                  className="flex-80"
                                >
                                  Upload Train Schedules Sheet
                                </p>
                                <FileUploader
                                  dispatchFn={[Function]}
                                  formClasses=""
                                  size="small"
                                  square={false}
                                  t={[Function]}
                                  text="Train Schedules .xlsx"
                                  theme={
                                    Object {
                                      "colors": Object {
                                        "primary": "#333",
                                        "secondary": "#fafafa",
                                      },
                                    }
                                  }
                                  tooltip=""
                                  type="xlsx"
                                  uploadFn={null}
                                />
                              </div>
                              <div
                                className="action_section flex-100 layout-row layout-align-center-center layout-wrap"
                              >
                                <p
                                  className="flex-80"
                                >
                                  Upload Vessel Schedules Sheet
                                </p>
                                <FileUploader
                                  dispatchFn={[Function]}
                                  formClasses=""
                                  size="small"
                                  square={false}
                                  t={[Function]}
                                  text="Vessel Schedules .xlsx"
                                  theme={
                                    Object {
                                      "colors": Object {
                                        "primary": "#333",
                                        "secondary": "#fafafa",
                                      },
                                    }
                                  }
                                  tooltip=""
                                  type="xlsx"
                                  uploadFn={null}
                                />
                              </div>
                              <div
                                className="action_section flex-100 layout-row layout-align-center-center layout-wrap"
                              >
                                <p
                                  className="flex-80"
                                >
                                  Upload Trucking Schedules Sheet
                                </p>
                                <FileUploader
                                  dispatchFn={[Function]}
                                  formClasses=""
                                  size="small"
                                  square={false}
                                  t={[Function]}
                                  text="Truck Schedules .xlsx"
                                  theme={
                                    Object {
                                      "colors": Object {
                                        "primary": "#333",
                                        "secondary": "#fafafa",
                                      },
                                    }
                                  }
                                  tooltip=""
                                  type="xlsx"
                                  uploadFn={null}
                                />
                              </div>
                            </div>
                          }
                          contentHeader=""
                          faClass="fa fa-cloud-upload"
                          handleCollapser={[Function]}
                          headerWrapClasses=""
                          hideIcon={false}
                          mainWrapperStyle={Object {}}
                          minHeight=""
                          parentClass=""
                          showArrow={true}
                          showArrowSpacer={false}
                          startCollapsed={false}
                          text="Upload Data"
                          theme={
                            Object {
                              "colors": Object {
                                "primary": "#333",
                                "secondary": "#fafafa",
                              },
                            }
                          }
                          wrapperContentClasses=""
                        />
                        <CollapsingBar
                          collapsed={true}
                          content={
                            <div>
                              <div
                                className="action_section flex-100 layout-row layout-wrap layout-align-center-center"
                              >
                                <p
                                  className="flex-100"
                                >
                                  Download Schedules Sheet
                                </p>
                                <DocumentsDownloader
                                  disabled={false}
                                  documentDispatch={null}
                                  downloadUrls={Object {}}
                                  loading={false}
                                  options={Object {}}
                                  size="small"
                                  square={false}
                                  t={[Function]}
                                  target="schedules"
                                  theme={
                                    Object {
                                      "colors": Object {
                                        "primary": "#333",
                                        "secondary": "#fafafa",
                                      },
                                    }
                                  }
                                  tooltip=""
                                />
                              </div>
                            </div>
                          }
                          contentHeader=""
                          faClass="fa fa-cloud-download"
                          handleCollapser={[Function]}
                          headerWrapClasses=""
                          hideIcon={false}
                          mainWrapperStyle={Object {}}
                          minHeight=""
                          parentClass=""
                          showArrow={true}
                          showArrowSpacer={false}
                          startCollapsed={false}
                          text="Download Data"
                          theme={
                            Object {
                              "colors": Object {
                                "primary": "#333",
                                "secondary": "#fafafa",
                              },
                            }
                          }
                          wrapperContentClasses=""
                        />
                        <CollapsingBar
                          collapsed={true}
                          content={
                            <div
                              className="action_section flex-100 layout-row layout-align-center-center layout-wrap"
                            >
                              <div
                                className="flex-none layout-row"
                              >
                                <RoundButton
                                  active={true}
                                  back={false}
                                  classNames=""
                                  disabled={false}
                                  flexContainer="100"
                                  handleDisabled={null}
                                  handleNext={[Function]}
                                  icon=""
                                  iconClass="fa-plus"
                                  inverse={false}
                                  size="small"
                                  text="New"
                                  theme={
                                    Object {
                                      "colors": Object {
                                        "primary": "#333",
                                        "secondary": "#fafafa",
                                      },
                                    }
                                  }
                                />
                              </div>
                            </div>
                          }
                          contentHeader=""
                          faClass="fa fa-plus-circle"
                          handleCollapser={[Function]}
                          headerWrapClasses=""
                          hideIcon={false}
                          mainWrapperStyle={Object {}}
                          minHeight=""
                          parentClass=""
                          showArrow={true}
                          showArrowSpacer={false}
                          startCollapsed={false}
                          text="Create new schedules"
                          theme={
                            Object {
                              "colors": Object {
                                "primary": "#333",
                                "secondary": "#fafafa",
                              },
                            }
                          }
                          wrapperContentClasses=""
                        />
                      </div>
                    }
                    flexOptions=""
                    header="Data manager"
                  />,
                ]
              }
              t={[Function]}
              theme={
                Object {
                  "colors": Object {
                    "primary": "#333",
                    "secondary": "#fafafa",
                  },
                }
              }
              toggleNew={null}
              user={
                Object {
                  "company_name": "FOO_COMPANY",
                  "currency": "EUR",
                  "email": "foo@bar.baz",
                  "first_name": "John",
                  "guest": false,
                  "internal": true,
                  "last_name": "Doe",
                  "phone": "6345789",
                  "role": Object {
                    "name": "shipper",
                  },
                }
              }
            />
          </Tab>
        </Tabs>
      </div>
    </div>
  </div>
</GenericError>
`;
